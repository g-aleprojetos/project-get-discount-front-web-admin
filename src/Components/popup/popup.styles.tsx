import styled from 'styled-components';
import colors from 'resources/colors';
import {TextH2} from 'Components/text';
import sizes from 'resources/sizes';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faXmark} from '@fortawesome/free-solid-svg-icons';

export const Cabecalho = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: ${sizes.large}px;
  border-top-left-radius: 13px;
  border-top-right-radius: 13px;
`;

export const Container = styled.div`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: ${colors.black50per};
`;

export const PopupBotao = styled.div`
  display: flex;
  margin-top: ${sizes.small}px;
`;

export const PopupBox = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-bottom: 13px;
  border-radius: 13px;
  background-color: ${colors.white};
`;

export const Principal = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-left: 21px;
  padding-right: 21px;
  gap: ${sizes.xsmall}px;
`;

export const Texto = styled(TextH2).attrs({
  cursor: 'pointer',
})``;

export const iconeFechar = styled(FontAwesomeIcon).attrs({
  icon: faXmark,
  height: 32,
  width: 32,
  size: 'xl',
})`
  margin-right: ${sizes.xsmall}px;
  margin-top: ${sizes.xxsmall}px;
  cursor: pointer;
`;

export const Icone = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
`;
