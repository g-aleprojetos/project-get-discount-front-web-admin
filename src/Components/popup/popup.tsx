import React, {useCallback} from 'react';
import {Button} from 'Components/button';
import {Message} from 'Components/message';
import {PropsButton} from 'Components/button/button.styles';
import * as S from './popup.styles';

type BotaoProps = {
  legenda?: string;
  aoClicar?: () => void | Promise<void>;
  corTexto?: string;
  corBotao?: string;
  negrito?: boolean;
  tipo?: PropsButton['tipo'];
};

export type Props = {
  testId?: string;
  textTitle?: string;
  textMessage?: string;
  corTextoTitulo?: string;
  corTextoMensagem?: string;
  botaoConfirmar?: BotaoProps;
  botaoCancelar?: BotaoProps;
  botaoOk?: BotaoProps;
  icone?: React.ReactNode;
  children?: React.ReactNode;
  aoSair: () => void | Promise<void>;
};

export const Popup = (props: Props) => {
  const {
    testId,
    textTitle,
    textMessage,
    corTextoTitulo,
    corTextoMensagem,
    botaoConfirmar,
    botaoCancelar,
    botaoOk,
    icone,
    children,
    aoSair,
  } = props;

  const obterPesoFonte = (negrito?: boolean) => {
    return negrito ? 'bold' : 'medium';
  };

  const handleAoSair = useCallback(() => {
    aoSair();
  }, [aoSair]);

  return (
    <S.Container data-testid={testId ?? 'popup-padrao'}>
      <S.PopupBox>
        <S.Cabecalho>
          <S.iconeFechar data-testid={'icone-fechar'} onClick={handleAoSair} />
        </S.Cabecalho>
        {icone && <S.Icone data-testid={'icone-cabecalho'}>{icone}</S.Icone>}
        <S.Principal>
          <Message
            textTitle={textTitle}
            textMessage={textMessage}
            colorTitle={corTextoTitulo}
            colorMessege={corTextoMensagem}
          />
          {children}
          {botaoOk ? (
            <S.PopupBotao>
              <Button
                testId="botao-ok"
                tipo={botaoOk.tipo ?? 'ok'}
                aoPressionar={botaoOk.aoClicar}>
                <S.Texto
                  cor={botaoOk?.corTexto}
                  peso={obterPesoFonte(Boolean(botaoOk?.negrito))}>
                  {botaoOk.legenda}
                </S.Texto>
              </Button>
            </S.PopupBotao>
          ) : (
            <S.PopupBotao>
              <Button
                testId="botao-cancelar"
                tipo={botaoCancelar?.tipo ?? 'cancelar'}
                cor={botaoCancelar?.corBotao}
                aoPressionar={handleAoSair}>
                <S.Texto
                  cor={botaoCancelar?.corTexto}
                  peso={obterPesoFonte(Boolean(botaoCancelar?.negrito))}>
                  {botaoCancelar?.legenda}
                </S.Texto>
              </Button>
              <Button
                testId="botao-confirmar"
                tipo={botaoConfirmar?.tipo ?? 'confirmar'}
                cor={botaoConfirmar?.corBotao}
                aoPressionar={botaoConfirmar?.aoClicar}>
                <S.Texto
                  cor={botaoConfirmar?.corTexto}
                  peso={obterPesoFonte(Boolean(botaoConfirmar?.negrito))}>
                  {botaoConfirmar?.legenda}
                </S.Texto>
              </Button>
            </S.PopupBotao>
          )}
        </S.Principal>
      </S.PopupBox>
    </S.Container>
  );
};
