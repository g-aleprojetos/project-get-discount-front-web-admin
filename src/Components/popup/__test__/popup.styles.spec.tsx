import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../popup.styles';

describe('popup.styles', () => {
  test('Cabecalho DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Cabecalho />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('PopupBotao  DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.PopupBotao />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('PopupBox DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.PopupBox />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Principal DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Principal />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Texto />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('iconeFechar DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.iconeFechar />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
