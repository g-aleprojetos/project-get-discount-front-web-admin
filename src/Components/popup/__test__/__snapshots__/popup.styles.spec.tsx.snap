// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup.styles Cabecalho DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 32px;
  border-top-left-radius: 13px;
  border-top-right-radius: 13px;
}

<div
  className="c0"
/>
`;

exports[`popup.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: #00000080;
}

<div
  className="c0"
/>
`;

exports[`popup.styles PopupBotao  DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 16px;
}

<div
  className="c0"
/>
`;

exports[`popup.styles PopupBox DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 13px;
  border-radius: 13px;
  background-color: #FFFFFF;
}

<div
  className="c0"
/>
`;

exports[`popup.styles Principal DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 21px;
  padding-right: 21px;
  gap: 8px;
}

<div
  className="c0"
/>
`;

exports[`popup.styles Texto DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 18px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 26px;
  cursor: pointer;
}

<h2
  className="c0 "
  cursor="pointer"
/>
`;

exports[`popup.styles iconeFechar DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin-right: 8px;
  margin-top: 4px;
  cursor: pointer;
}

<svg
  aria-hidden="true"
  className="svg-inline--fa fa-xmark fa-xl c0"
  data-icon="xmark"
  data-prefix="fas"
  focusable="false"
  height={32}
  role="img"
  style={Object {}}
  viewBox="0 0 384 512"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
